src/app/core/tasks/threadpool.nim

```diff
@@ -1,260 +1,59 @@
-import # std libs
-  atomics, json, sequtils, tables
-
 import # vendor libs
-  chronicles, chronos, json_serialization, task_runner
+  json_serialization, json, chronicles, taskpools
 
 import # status-desktop libs
-  ./common
+  ./common 
 
-export
-  chronos, common, json_serialization
+export common, json_serialization
 
 logScope:
   topics = "task-threadpool"
 
 type
   ThreadPool* = ref object
-    chanRecvFromPool: AsyncChannel[ThreadSafeString]
-    chanSendToPool: AsyncChannel[ThreadSafeString]
-    thread: Thread[PoolThreadArg]
-    size: int
-    running*: Atomic[bool]
-  PoolThreadArg = object
-    chanSendToMain: AsyncChannel[ThreadSafeString]
-    chanRecvFromMain: AsyncChannel[ThreadSafeString]
-    size: int
-  TaskThreadArg = object
-    id: int
-    chanRecvFromPool: AsyncChannel[ThreadSafeString]
-    chanSendToPool: AsyncChannel[ThreadSafeString]
-  ThreadNotification = object
-    id: int
-    notice: string
-
-# forward declarations
-proc poolThread(arg: PoolThreadArg) {.thread.}
+    pool: Taskpool
+  ThreadSafeTaskArg* = distinct cstring
 
 const MaxThreadPoolSize = 16
```

Comment:
Just wondering... why is this hardcoded to 16? What about CPUs with less threads?

Reply:
Agreed, we should get the user's number of cores. the `os` lib should have access to that.